docs/code-review.md

```diff
@@ -0,0 +1,363 @@
+Tests should fail with helpful messages saying what was wrong, what inputs were provided, the expected result, and the actual result received in testing. Assume that the person debugging your failing test is not you, and is potentially new to the entire project.
+
+#### Unit tests
+
+The code should be testable by unit functions, which means that each function should perform an action and this action should be testable.
+
+When creating unit tests, try to use table-driven tests when the same result is expected (the function must return an error) for different inputs. Table-driven tests allow us to create a boilerplate in which we define the name of the test (explanation of what it does), the test inputs and the output to be returned. For example:
+
+```golang
+func Test_FunctionNameWithNoError(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+    }{
+        {
+            name:        "when resource is nil then error",
+            resource:    nil,
+        },
+        {
+            name:        "when resource is not valid then error",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource is valid then no error",
+            resource:    CorrectResource{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            assert.NoError(t, err)
+        })
+    }
+}
+
+func Test_FunctionNameWithError(t *testing.T) {
+    t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+    testDir := t.TempDir()
+    err := functionName(MalformedResource{})
+    assert.NoError(t, err)
+}
+```
+
+If the test has many use cases, it will be grouped into several tests per result, for example:
+
+```golang
+func Test_FunctionNameThenErr(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+        expectedErr  bool
+    }{
+        {
+            name:        "when resource is nil then error",
+            resource:    nil,
+        },
+        {
+            name:        "when resource is not valid then error",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource is not valid because of X",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource fails because of proxy",
+            resource:    MalformedResource{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            if tt.expectedErr {
+                assert.Error(t, err)
+            } else {
+                assert.NoError(t, err)
+            }
+        })
+    }
+}
```

Comment:
I wouldn't include an example with a non-recommended approach for unit testing. Including conditionals in the test to verify some conditions or others is error prone.

Having said that, I think this example doesn't really want to use the conditional. I think we can remove the `expectedErr` as part of the test input and remove the conditional in the line 137 to only verify `assert.Error(t, err)`. All input cases are expecting an error